### ISSUE TITLE
Message control fails when adding a message for a node in a different file

### ISSUE THREAD

**Issue by xmo-odoo** [PyCQA/pylint#8066]
### Bug description

When implementing a lint with multiple "passes", one of which creates a message for the other (with nodes stored by the first pass rather than its own), message control (`# pylint: disable` strictures) seem to work fine if the two passes hit in the same file, but don't if the second pass hits in a different file than the first.

Checker:
```python
from collections import defaultdict

import astroid
from pylint.checkers import BaseChecker
from pylint.checkers.utils import only_required_for_messages
from pylint.lint import PyLinter


class Checker(BaseChecker):
    msgs = {
        'E9999': ("message", 'message', '???')
    }

    def __init__(self, linter: PyLinter) -> None:
        super().__init__(linter)
        self.calls = defaultdict(list)

    @only_required_for_messages('message')
    def visit_call(self, node):
        if node.func.name not in self.defs:
            self.calls[node.func.name].append(node)

    @only_required_for_messages('message')
    def visit_functiondef(self, node):
        for call in self.calls.pop(node.name, []):
            self.add_message("message", node=call)

def register(linter):
    linter.register_checker(Checker(linter))
```
When this checker hits a function call it remembers the function, then when it hits the corresponding definition it adds a message *for the function call*.

Test files:
```python
# a.py
from b import *

def qux():
    quux()  # [message]
    corge()  # [message]

def qux2():
    # pylint: disable=message
    quux()
    corge()
```
```python
# b.py
def quux():
    ...

def corge():
    ...
```
here even though they should be disabled the two calls in `qux2` are flagged. If the content of `b.py` is moved to the bottom of `a.py`, then the behaviour is as expected (only the callsites in `qux` are flagged).

### Command used

```shell
pylint --disable=all --enable=message --load-plugins=test_plugin a.py b.py
```

### Pylint output

```shell
************* Module a
a.py:4:4: E9999: message (message)
a.py:9:4: E9999: message (message)
a.py:5:4: E9999: message (message)
a.py:10:4: E9999: message (message)
```

### Expected behavior

************* Module a
a.py:4:4: E9999: message (message)
a.py:5:4: E9999: message (message)

### Pylint version

```shell
pylint 2.15.8
astroid 2.12.13
Python 3.8.13 (default, Aug 10 2022, 11:04:47)
```

### OS / Environment

Mint 21, kernel 5.15.0-56

**Comment by DanielNoord** [comment]
This is unsupported sadly. We don't guarantee that methods of checkers that depend on each other can do so.

**Comment by DanielNoord** [comment]
You could maybe hack it in by using `close()`, but that only gets called after all files have been visited so you would need to keep a much larger state object on the checker. We use this ourselves twice in the codebase (just grep for `def close`).

Other than that I can't really think of one.

I'm tagging @jacobtylerwalls as they know much more about the "unused" checker which I think faces a similar problem. They might be able to help out here.
